drivers/input/input_pinnacle.c

```diff
@@ -0,0 +1,627 @@
+/*
+ * Copyright (c) 2024 Ilia Kharin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT cirque_pinnacle
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+#include <zephyr/drivers/i2c.h>
+#endif
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#include <zephyr/drivers/spi.h>
+#endif
+#include <zephyr/init.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/util_macro.h>
+
+LOG_MODULE_REGISTER(PINNACLE, CONFIG_INPUT_LOG_LEVEL);
+
+/*
+ * Register Access Protocol Standard Registers.
+ * Standard registers have 5-bit addresses, BIT[4:0], that range from
+ * 0x00 to 0x1F. For reading, a register address has to be combined with
+ * 0xA0 for reading and 0x80 for writing bits, BIT[7:5].
```

Comment:
is that true for i2c as well?

Reply:
@fabiobaltieri, yes, in my understanding this is true for both SPI and I2C.